VIPs/vip-7.md

```diff
@@ -0,0 +1,260 @@
+
+## Node operator
+
+It may be preferable for Users to have nodes operated by different operators. ie: You may not want all your nodes operated by a particular organization.
+
+## User preference
+
+Users may have specific preferences. Here are some examples:
+
+1. I only care about having the fastest possible experience
+2. I don't trust my government, so want to ensure my data is stored in another country, regardless of how slow it is
+3. I want to maximise redundancy, so want to choose storage nodes across 5 different countries
+
+The protocol will focus on providing sensible default for all users, but it will be possible for advanced users to completely customize where their data is stored.
+
+# Specification
+
+## Discovery
+
+### Storage Nodes
+
+The list of verified storage node operators will be maintained in a JSON file at a known URL (`Storage Node Registry`). This URL endpoint will be updated by the Verida Foundation and embedded in the Verida protocol client implementation.
+
+This file will be located at:
+
+1. https://meta.verida.network/registry/storageNodes/testnet.json
+2. https://meta.verida.network/registry/storageNodes/mainnet.json
+
+Example file:
+
+```
+[
+    {
+        "id": "verida-testnet-aws-us-east1-001",
+        "name": "Verida Foundation Testnet: AWS-US-EAST1-001",
+        "description": "Node operated by the Verida Foundation on Testnet",
+        "datacenter": "aws-us-east1",
+        "serviceEndpoint": "https://001-use1.tn.verida.tech/",
+        "establishmentDate": "2023-01-03T08:22:35Z",
+        "countryResidence": "SG"
+    },
+    ...
+]
+```
+
+### Data Centers
+
+The list of official data centers will be maintained in a JSON file at a known URL (`Datacenter Registry`). This URL endpoint will be updated by the Verida Foundation and embedded in the Verida protocol client implementation.
+
+This file will be located at:
+
+1. https://meta.verida.network/registry/dataCenters/testnet.json
+2. https://meta.verida.network/registry/dataCenters/mainnet.json
+
+Example file:
+
+```
+[
+    {
+        "id": "aws-us-east1",
+        "name": "Amazon Web Services: US East1 (North Virginia)",
+        "countryLocation": "US",
+        "latitude": 38.9809166,
+        "longitude": -77.4692733,
+    },
+    ...
+]
+```
+
+### Regions
+
+The list of official regions and their associated countries will be maintained in a JSON file at a known URL (`Datacenter Registry`). This URL endpoint will be updated by the Verida Foundation and embedded in the Verida protocol client implementation.
+
+This file will be located at:
+
+1. https://meta.verida.network/registry/regions/testnet.csv
+2. https://meta.verida.network/registry/regions/mainnet.csv
+
+Country codes will be two character alpha representations of the country. Regions will be fixed strings.
```

Comment:
two character ISO codes.

See previous comments about regions

Reply:
Specification section updated to be clearer.